backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt

```diff
@@ -1584,8 +1584,31 @@ internal class CodeGeneratorVisitor(val context: Context, val lifetimes: Map<IrE
         return !irClass.isFrozen
     }
 
+    private fun isDefaultValue(value: IrExpression): Boolean {
+        if (value !is IrConst<*>) return false
+        return when (value.kind) {
+            IrConstKind.Null -> true
+            IrConstKind.Boolean -> (value.value as Boolean) == false
+            IrConstKind.Byte -> (value.value as Byte) == 0.toByte()
+            IrConstKind.Char -> (value.value as Char) == 0.toChar()
+            IrConstKind.Short -> (value.value as Short) == 0.toShort()
+            IrConstKind.Int -> (value.value as Int) == 0
+            IrConstKind.Long -> (value.value as Long) == 0L
+            IrConstKind.Float -> (value.value as Float) == 0.0f
+            IrConstKind.Double -> (value.value as Double) == 0.0
```

Comment:
Double and Float might require specific handling here, because `==` is not a bitwise comparison for these time, so can return `true` for a value that is different from all bits set to 0. See also support for this behaviour in JVM backends.

Reply:
You mean some corner cases like +0.0 == -0.0? I forgot about them.

Reply:
Yes.